source/agora/test/PreImageCatchup.d

```diff
@@ -0,0 +1,64 @@
+/*******************************************************************************
+
+    Check that a validator can retrieve missing pre-images when it could not
+    receive pre-images through `postPreimage`. We altered the algorithm of
+    distributing pre-images so that a validator does not continuously send
+    pre-images any more without checking that the previous pre-image is
+    posted normally. So the network should survive the situation where
+    some pre-images are missing.
```

Comment:
Small DDOC note: The title/summary should be short, and the longer explanation should follow.
E.g.
```D
/*******************************************************************************

    Some high level overview about this, in a single sentence, if possible.

    A longer explanation that can span as many sentences as one want.
    And obviously multiple line. Give as much details as you wish, really.

    Specific_notes:
      For specific topic, it's possible to have sub-heading using the same
      approach as `Params` and `Returns`. In fact, `Params`, `Returns`,
      and the likes are simply recognized sub heading and don't need to come
      in a specific order.

*******************************************************************************/
```

Reply:
Simplified the summary of the unit test because the goal of the test has been made clearer.